src/HealthChecks/Abstractions/src/HealthCheckRegistration.cs

```diff
@@ -79,10 +88,16 @@ public HealthCheckRegistration(string name, IHealthCheck instance, HealthStatus?
                 throw new ArgumentNullException(nameof(factory));
             }
 
+            if (timeout.HasValue && timeout.Value <= TimeSpan.Zero && timeout.Value != System.Threading.Timeout.InfiniteTimeSpan)
```

Comment:
Can you add a using for `System.Threading`?

Reply:
Nope, name collision with the property :/